core/src/cgcloud/core/box.py

```diff
@@ -406,8 +524,34 @@ def create( self, wait_ready=True, cluster_ordinal=0 ):
                 result.append( box )
         except StopIteration:
             pass
+
         return result
 
+    def _spot_create(self):
+        """
+        A request for multiple instances actually creates many requests for 1 instance each, which then become
+        reservations with 1 instance each.
+        :return: list of boto.ec2.instance.Reservation objects
+        """
+        requests = self.ctx.ec2.request_spot_instances(image_id=self.image_id, **self.instance_creation_args)
+        instances = []
+        try:
+            for request in requests:
+                while True:
+                    # get_all_spot_instances_requests returns a list, even if you request only 1 id like in this case
+                    updatedRequest = unpack_singleton(self.ctx.ec2.get_all_spot_instance_requests(request_ids=[request.id]))
```

Comment:
This is very inefficient. Why do we have to do this one instance at a time? Can't we query the status of all requests in a single round-trip?